### PR TITLE
New web steps for tabs and frames.

### PR DESCRIPTION
**Pull request by xsmrcek**
FORCE_CLICK web action.
Documentation for web steps amended.

### CODE REVIEW

**Comment by paveljandejsek** [comment]
Generated [reports](http://95.168.205.146:8080/reports/127/index.xml)

File: jbehave-support-core-test/jbehave-support-core-test-app/src/main/resources/static/title.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE HTML>
```
Comment (paveljandejsek) [inline jbehave-support-core-test/jbehave-support-core-test-app/src/main/resources/static/title.html]: Nitpicking: maybe change the name from title.html to something like frame.html or iframe.html?

Comment (xsmrcek) [reply jbehave-support-core-test/jbehave-support-core-test-app/src/main/resources/static/title.html]: renamed to frames.html and title changes to Marvelous Frames

File: jbehave-support-core/src/main/java/org/jbehavesupport/core/web/WebSteps.java [hunk 2]
```diff
@@ -193,6 +205,52 @@ public void switchToNewTab() {
...
+                handlesAfter.removeAll(handlesBefore);
+                driver.switchTo().window(handlesAfter.iterator().next());
+                return;
+            }
+        }
+        throw new AssertionError("Opening new tab failed");
+    }
+
+    @Then("tab with [$urlTitle] containing [$text] is focused")
+    public void findTabWithUrl(String urlTitle, ExpressionEvaluatingParameter<String> text) {
```
[the remark this answers is not in the document]
Comment (xsmrcek) [reply jbehave-support-core/src/main/java/org/jbehavesupport/core/web/WebSteps.java]: renamed

File: jbehave-support-core/src/main/java/org/jbehavesupport/core/web/WebSteps.java [hunk 3]
```diff
@@ -185,6 +192,11 @@ public void waitUntilCondition(String page, String element, ExpressionEvaluating
         waitCondition.evaluate(waitConditionCtx);
     }
 
+    /**
+     * @deprecated because it's not opening new tab, only focus tab opened by another action
```
[the remark this answers is not in the document]
Comment (xsmrcek) [reply jbehave-support-core/src/main/java/org/jbehavesupport/core/web/WebSteps.java]: Added

File: jbehave-support-core/docs/Web-testing.md [hunk 4]
```diff
@@ -142,6 +142,33 @@ When navigated forward
...
 
+To focus iframes and return to main frame.
+```
+Given on page [home] frame [iframe] is focused
+```
+```
+Given main frame is focused
+```
+
+To open a new tab.
```
[the remark this answers is not in the document]
Comment (xsmrcek) [reply jbehave-support-core/docs/Web-testing.md]: Added

File: jbehave-support-core/docs/Web-testing.md [hunk 5]
```diff
@@ -161,6 +188,7 @@ There are the following actions available at the moment:
 - DISMISS - for dismissing alert dialog
 - CLICK
 - DOUBLE_CLICK
+- FORCE_CLICK - for clicking on enabled but invisible elements
```
[the remark this answers is not in the document]
Comment (xsmrcek) [reply jbehave-support-core/docs/Web-testing.md]: Added

**Comment by paveljandejsek** [comment]
Generated [reports](http://95.168.205.146:8080/reports/127/index.xml)

**Comment by xsmrcek** [comment]
@thradec Given annotation added.

**Comment by paveljandejsek** [comment]
Generated [reports](http://95.168.205.146:8080/reports/127/index.xml)

**Pull request merged**